bee-bundle/src/transaction.rs

```diff
@@ -404,6 +451,85 @@ impl Transaction {
         Ok(transaction)
     }
 
+    fn into_trits_allocated(&self, mut buf: &mut Trits<T1B1>) {
+        buf.copy_raw_bytes(
+            self.payload().to_inner(),
+            PAYLOAD.trit_offset.start,
+            PAYLOAD.trit_offset.length,
+        );
+        buf.copy_raw_bytes(
+            self.address().to_inner(),
+            ADDRESS.trit_offset.start,
+            ADDRESS.trit_offset.length,
+        );
+        buf.copy_raw_bytes(
+            self.obsolete_tag().to_inner(),
+            OBSOLETE_TAG.trit_offset.start,
+            OBSOLETE_TAG.trit_offset.length,
+        );
+
+        buf.copy_raw_bytes(
+            self.bundle().to_inner(),
+            BUNDLE.trit_offset.start,
+            BUNDLE.trit_offset.length,
+        );
+
+        buf.copy_raw_bytes(
+            self.branch().to_inner(),
+            BRANCH.trit_offset.start,
+            BRANCH.trit_offset.length,
+        );
+
+        buf.copy_raw_bytes(
+            self.trunk().to_inner(),
+            TRUNK.trit_offset.start,
+            TRUNK.trit_offset.length,
+        );
+
+        buf.copy_raw_bytes(self.tag().to_inner(), TAG.trit_offset.start, TAG.trit_offset.length);
+
+        buf.copy_raw_bytes(
+            self.nonce().to_inner(),
+            NONCE.trit_offset.start,
+            NONCE.trit_offset.length,
+        );
+
+        let value_buf = TritBuf::<T1B1Buf>::try_from(self.value().to_inner().to_owned()).unwrap();
```

Comment:
Do these really need to take an owned valued ? This makes a copy every time :(

Reply:
for i64 it should be super fine but i guess for TritBuf we should impl Fom<&TritBuf>

Reply:
It's definitely fine unless you are on an arch that is not x64

Reply:
Why ?

Reply:
There are other `.to_owned()` btw :)

Reply:
sure :)
fixed though it's i64

Reply:
i just see you asked why, because copying is more expensive than moving only when the copied object is bigger than an address size, when it's i64, does it matter if it's copied or moved or borrowed?

Reply:
Yeah you're most probably right. Just find this Transaction API so verbose :(